### PR TITLE
basic ruby functionality 

### PR DESCRIPTION
**Pull request by mbobin**

### CODE REVIEW

File: minio/lib/minio/config.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    def initialize(args = {})
+      @endpoint   = args[:endpoint] || 'http://localhost:9000'
+      @port       = args[:port] || 9000
+      @access_key = args[:access_key]
+      @secret_key = args[:secret_key]
+      @secure     = args[:secure]
+      @transport  = args[:transport]
+      @region     = args[:region] || 'us-east-1'
+      @service    = args[:service] || 's3'
```
Comment (harshavardhana) [inline minio/lib/minio/config.rb]: service is not configurable since this is an S3 client.

File: minio/lib/minio/config.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module MinioRuby
+  class Config
+    attr_accessor :endpoint, :port, :transport, :secure
+    attr_accessor :access_key, :secret_key, :region, :service
+
+    def initialize(args = {})
+      @endpoint   = args[:endpoint] || 'http://localhost:9000'
```
Comment (harshavardhana) [inline minio/lib/minio/config.rb]: the endpoint shouldn't be of form `http://<hostname>:<port>` you should take endpoint of form `<hostname>:<port>` and avoid taking `<port>` as a special argument.

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Thanks a lot, wonderful changes. I have few comments..

**Review by harshavardhana — APPROVED** [review]

**Review by deekoder — APPROVED** [review]

**Comment by deekoder** [comment]
@mbobin great contribution, we'd love to send you a MinIO t-shirt. Could you come to our slack channel https://slack.min.io/

**Comment by mbobin** [comment]
Awesome! Thank you!

What would be the next steps? Other functionality?

**Pull request merged**